solr-ingest/src/main/java/edu/unc/lib/dl/data/ingest/solr/filter/SetDescriptiveMetadataFilter.java

```diff
@@ -253,6 +255,33 @@ private void extractAbstract(Element mods, IndexDocumentBean idb) {
         }
     }
 
+    private void extractCollectionId(Element mods, IndexDocumentBean idb) {
+        List<?> identifiers = mods.getChildren("identifier", JDOMNamespaceUtil.MODS_V3_NS);
+        List<String> collectionId = new ArrayList<>();
+
+        if (identifiers != null && identifiers.size() > 0) {
+            for (Object id: identifiers) {
+                Element aid = (Element) id;
+                Attribute type = aid.getAttribute("type");
+                Attribute collection = aid.getAttribute("displayLabel");
+
+                if (type == null || collection == null) {
+                    continue;
+                }
+
+                if (collection.getValue().equals("Collection Number") && type.getValue().equals("local")) {
+                    collectionId.add(aid.getValue());
+                }
+            }
+        }
+
+        if (collectionId.size() > 0) {
+            idb.setCollectionId(collectionId.get(0));
```

Comment:
if you only care about the first one, you could break out of the loop when it finds the collection number rather than continuing to look. In that case you wouldn't need a List either.